accord-core/src/main/java/accord/local/CommandStore.java

```diff
@@ -128,6 +128,8 @@ public RangesForEpochHolder rangesForEpochHolder()
 
     public abstract void shutdown();
 
+    public abstract AsyncChain<Timestamp> maxAppliedFor(Seekables<?, ?> keysOrRanges, Ranges slice);
```

Comment:
I feel this is best in `SafeCommandStore`, at least in the C* case you would need to do (logically)

```
return submit(PreloadContext.contextFor(keysOrRanges), safe -> safe.maxAppliedFor(keysOrRanges, Ranges slice);
```

Reply:
I don't agree. The safe store you'd be calling couldn't be used for maxApplied since it wouldn't have the keys in scope

Reply:
why would it not have the keys?  Isn't the job of `submit(PreLoadContext)` to make sure the keys are present (for key and ranges)?

I do think @belliottsmith didn't like me moving `maxConflict` to the safe store... but keeping it in `CommandStore` made C* more complex, as we don't have everything in-memory... so we are forced to push this logic into the executor (aka `SafeCommandStore`)